### PR TITLE
Adding the ability to pass a torchcraft state to setMap

### PR DESCRIPTION
**Pull request by ebetica**
Fixes https://github.com/TorchCraft/TorchCraft/issues/107.

### CODE REVIEW

File: include/replayer.h [hunk 1]
```diff
@@ -92,13 +93,23 @@ class Replayer : public RefCounted {
     return numUnits.at(key);
   }
 
-  void setMap(THByteTensor* walkability, THByteTensor* ground_height,
-      THByteTensor* buildability,
-      std::vector<int>& start_loc_x, std::vector<int>& start_loc_y);
+  void setMap(torchcraft::State* state);
```
Comment (jgehring) [inline include/replayer.h]: Maybe rename this to `setMapFromState` for clarity?

File: replayer/replayer.cpp [hunk 2]
```diff
@@ -161,6 +161,30 @@ void Replayer::setMap(
...
+    for (int x = 0; x < w; x++) {
+      uint8_t v_w = (uint8_t)(state->walkable_data[y * w + x]);
+      uint8_t v_b = (uint8_t)(state->buildable_data[y * w + x]);
+      // Ground height only goes up to 5
+      uint8_t v_g = state->ground_height_data[y * w + x] & 0b111;
+      uint8_t packed = (v_w << WALKABILITY_SHIFT) |
+          (v_b << BUILDABILITY_SHIFT) | (v_g << HEIGHT_SHIFT);
+      THTensor_fastSet2d(map.data, y, x, packed);
+    }
+  }
```
Comment (jgehring) [inline replayer/replayer.cpp]: This loop could be factored out with #109.

**Comment by jgehring** [comment]
Ok if I merge #109 and you rebase this afterwards?

**Comment by ebetica** [comment]
I did a rebase

File: lua/replayer_lua.cpp [hunk 3]
```diff
@@ -158,10 +168,10 @@ extern "C" int replayerSetMap(lua_State* L) {
       luaT_checkudata(L, 4, "torch.ByteTensor"));
   if (!lua_istable(L, 5))
     luaL_error(L, "bad argument #%d (argument must be table)", 5);
-  std::vector<int> start_loc_x, start_loc_y;
-  for (int i=1; ; i++) {
+  for (int i = 1;; i++) {
```
Comment (jgehring) [inline lua/replayer_lua.cpp]: Nit: the standard way to iterate over a Lua table is to use `lua_next()`.

File: replayer/replayer.cpp [hunk 4]
```diff
@@ -21,8 +21,8 @@ namespace replayer {
 
 std::ostream& operator<<(std::ostream& out, const Replayer& o) {
   auto height = THByteTensor_size(o.map.data, 0);
-  auto width = THByteTensor_size(o.map.data, 1);
-  auto data = THByteTensor_data(o.map.data);
+  auto width  = THByteTensor_size(o.map.data, 1);
+  auto data   = THByteTensor_data(o.map.data);
```
Comment (jgehring) [inline replayer/replayer.cpp]: :cry:

**Pull request merged**